azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -44,15 +47,19 @@
  * This servlet exposes the REST APIs such as create, get etc. for image type. Below are the
  * supported APIs. Create Image Type API: POST /imageTypes?session.id=? --data @payload.json Get
  * Image Type API: GET /imageTypes?session.id=?&imageType=? GET /imageTypes/{id}?session.id=?
+ * GET /imageTypes/{imageType}?sessionid=? GET /imageTypes/{imageType}?proxyUser=?&session.id=?
  */
```

Comment:
Thank you for the comment! My understanding is that this request will be sent via a python script invoked by Jenkins, and therefore will not go through a browser, and perhaps therefore not be cached? I guess it is bad to have that kind of optionality on a GET request, but for what we're trying to do Jenkins side, I'm not sure how we get around using a session id given the patterns we're currently using. I will try to avoid this kind of thing going forward.

Reply:
A browser is just one stop in the communication chain, there are DNS servers, proxies, etc. which can also cache the requests. 
Please remember Azkaban is open source, LinkedIn use cases should not be the only ones considered when implementing a change.

Reply:
I'm not suggesting to drop the use of a session id, I'm just saying that it should not be sent as a query parameter but as a cookie.

Reply:
Thank you for the clarification, thank makes sense!

Reply:
Agree with @ypadron-in . Let's have separate ticket to remove session id from query param. It might require us to inform internal API users as well as currently they might be using session id in query param for various APIs.